src/main/CommandLine.cpp

```diff
@@ -1281,6 +1281,7 @@ run(CommandLineArgs const& args)
                     cfg.MODE_STORES_HISTORY_MISC = false;
                     cfg.MODE_USES_IN_MEMORY_LEDGER = false;
                     cfg.MODE_ENABLES_BUCKETLIST = false;
+                    cfg.EXPERIMENTAL_BUCKETLIST_DB = false;
```

Comment:
Should we just have a single free function that determines if this flag should is true? The condition would be `EXPERIMENTAL_BUCKETLIST_DB && !MODE_USES_IN_MEMORY_LEDGER && MODE_ENABLES_BUCKETLIST`. Hard-coding it manually can lead to weird configurations (like the flag is on, but core doesn't enable bucketlist).

Reply:
We should probably update it here as well with a warning to the user: 
https://github.com/stellar/stellar-core/blob/master/src/main/Config.cpp#L980